File: Python/StockMonitor.py
# To find which day if we buy and sell the stock will 
# get maximum profit...
def sellDay(price):
    maxProfit = 0
    minPrice = float('inf')
    for i in range(len(price)):
        if (price[i] < minPrice):
            minPrice = price[i]
        elif price[i] - minPrice > maxProfit:
            maxProfit = price[i] - minPrice
    return maxProfit
        
price = [7, 4, 1, 3, 6, 16]
result = sellDay(price)
print(result)

File: Java/StockMonitor.java
public class StockMonitor {

    public static double findMaxProfit(int arr[]) {
        double maxProfit = 0;
        double minPrice = Double.MAX_VALUE;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] < minPrice) {
                minPrice = arr[i];
            } else if (arr[i] - minPrice > maxProfit) {
                maxProfit = arr[i] - minPrice;
            }
        }

        return maxProfit;
    }

    public static void main(String[] args) {
        int arr[] = { 7, 4, 5, 1, 6, 16};
        double result = findMaxProfit(arr);
        System.out.println(result);
    }
}
